src/main/java/luke/task/TaskList.java

```diff
@@ -0,0 +1,170 @@
+     */
+    public String addToList(String command, String taskDetails, boolean isMarked, boolean isLoadingFromDisk)
+            throws NoDescriptionException, UnknownCommandException {
+        if (!(Constants.TASK_TYPES.contains(command))) {
+            throw new UnknownCommandException();
+        }
+        if (taskDetails.isEmpty()) {
+            throw new NoDescriptionException();
+        }
+        switch (command) {
+        case "todo" -> {
+            return addTodoToList(taskDetails, isMarked, isLoadingFromDisk);
+        }
+        case "deadline" -> {
+            return addDeadlineToList(taskDetails, isMarked, isLoadingFromDisk);
+        }
+        case "event" -> {
+            return addEventToList(taskDetails, isMarked, isLoadingFromDisk);
+        }
+        default -> {
+            throw new UnknownCommandException();
+        }
+        }
+    }
+
+    private String addTodoToList(String taskDetails, boolean isMarked, boolean isLoadingFromDisk) {
+        Todo todo = new Todo(taskDetails, isMarked);
+        taskList.add(todo);
+        if (!isLoadingFromDisk) {
+            return "i've thrown this to-do into your task list:\n"
+                    + Constants.INDENT + todo.taskDescription() + "\n"
+                    + listSizeUpdateMessage();
+        }
+        return "";
+    }
+
+    private String addDeadlineToList(String taskDetails, boolean isMarked, boolean isLoadingFromDisk) {
+        String[] taskAndDeadline;
+        if (isLoadingFromDisk) {
+            taskAndDeadline = taskDetails.split(" by ");
+        } else {
+            taskAndDeadline = taskDetails.split(" /by ");
+        }
+        String taskName = taskAndDeadline[0];
+        String deadline = taskAndDeadline[1];
+        Deadline dl = new Deadline(taskName, deadline, isMarked);
+        addTask(dl);
+        if (!isLoadingFromDisk) {
+            return "the new deadline's been added to your task list:\n"
+                    + Constants.INDENT + dl.taskDescription() + "\n"
+                    + listSizeUpdateMessage();
+        }
+        return "";
+    }
+
+    private String addEventToList(String taskDetails, boolean isMarked, boolean isLoadingFromDisk) {
+        String[] taskAndTimings;
+        if (isLoadingFromDisk) {
+            taskAndTimings = taskDetails.split(" from | to ");
+        } else {
+            taskAndTimings = taskDetails.split(" /from | /to ");
+        }
+        String taskName = taskAndTimings[0];
+        String from = taskAndTimings[1];
+        String to = taskAndTimings[2];
+        Event event = new Event(taskName, from, to, isMarked);
+        addTask(event);
+        if (!isLoadingFromDisk) {
+            return "aaaaand this event is now in your task list:\n"
+                    + Constants.INDENT + event.taskDescription() + "\n"
+                    + listSizeUpdateMessage();
+        }
+        return "";
+    }
```

Comment:
Maybe can extract the function for parsing data to their respective task class. For example you can parse the input into two parts, the task type and the description, and process the description for deadline and event task to split the task description and timing within the class constructor. Just a small suggestion!